src/tdlib/chat_history.rs

```diff
@@ -145,7 +148,10 @@ impl ChatHistory {
             }
             MessageContent(ref update_) => self.handle_message_update(update_.message_id, update),
             MessageEdited(ref update_) => self.handle_message_update(update_.message_id, update),
-            MessageSendSucceeded(update) => self.remove(update.old_message_id),
+            MessageSendSucceeded(update) => {
+                self.remove(update.old_message_id);
+                self.push_front(update.message);
+            }
```

Comment:
Mh, this is causing a flicker, but that can probably be solved if we call `items_changed` once.

Reply:
Well, nevermind, we can't call `items_changed` once because it needs an index and in this case we can't be sure that the position of the old message will be the same as the new one (some other messages may have been sent meanwhile).

Reply:
So, I consider the flicker a GTK bug, since API-wise we can't do anything different.

Reply:
From my experiments, it seems solvable if we don't exchange the whole message, but only update the properties of the old message.
But that's a different story.

Reply:
The problem is that we still need to update the position since it may have changed, and to do that we still need to call `items_changed` twice (one to notify the removal of the previous one and one for the new position).